src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/Common/WebRequestPSCmdlet.Common.cs

```diff
@@ -1283,17 +1280,15 @@ internal virtual HttpResponseMessage GetResponse(HttpClient client, HttpRequestM
                     _cancelToken.Cancel();
                     _cancelToken = null;
 
-                    // if explicit count was provided, reduce it for this redirection.
+                    // If explicit count was provided, reduce it for this redirection.
                     if (WebSession.MaximumRedirection > 0)
                     {
                         WebSession.MaximumRedirection--;
                     }
-                    // For selected redirects that used POST, GET must be used with the
-                    // redirected Location.
-                    // Since GET is the default; POST only occurs when -Method POST is used.
-                    if (Method == WebRequestMethod.Post && IsRedirectToGet(response.StatusCode))
+
+                    // For selected redirects, GET must be used with the redirected Location.
+                    if (RequestRequiresForceGet(response.StatusCode, req.Method) && !PreserveHttpMethodOnRedirect)
                     {
-                        // See https://msdn.microsoft.com/library/system.net.httpstatuscode(v=vs.110).aspx
                         Method = WebRequestMethod.Get;
```

Comment:
@iSazonov I think I found an edge case bug:
if we get to this point with `-CustomMethod "POST"`, setting `Method = WebRequestMethod.Get` would be ignored by `HttpMethod httpMethod = string.IsNullOrEmpty(CustomMethod) ? GetHttpMethod(Method) : new HttpMethod(CustomMethod);`.
I have two solutions to this problem:
- Easy Fix -> Set `CustomMethod = null` in this if { }
- Breaking Change -> Remove CustomMethod (I know it's public) and just use `new  HttpMethod(whatever)`

This Bug should also be present in pwsh 7.3.2 but I don't know how to test it, and it's easier to fix it in this PR

Reply:
> HttpMethod httpMethod = string.IsNullOrEmpty(CustomMethod) ? GetHttpMethod(Method) : new HttpMethod(CustomMethod);

I can not find where this code is.

Reply:
In `GetRequest` line 996

Reply:
I will fix it in another PR.
I might be wrong but i think the bug was already there before #18846.

Reply:
You could check this with Fiddler :-)

Reply:
Added tests in #19111

Reply:
@iSazonov should I also move the reworked `IsRedirectCode`, `RequestRequiresForceGet` and `ShouldRetry` to another PR to merge them sooner?

Reply:
Main stopper is absence of WG approve for new parameter.